packages/react-dom/src/__tests__/DOMPropertyOperations-test.js

```diff
@@ -157,6 +157,21 @@ describe('DOMPropertyOperations', () => {
     });
   });
 
+  it('should warn if attributes take too long to stringify', () => {
+    const container = document.createElement('div');
+    const attributeValue = {foo: 'bar'};
+    attributeValue.toString = function() {
+      // for loop to waste time
+      for (let i = 0; i < 54321; i++) {}
```

Comment:
Actually this does not waste time because the loop does nothing. Most JavaScript engines detect this loop as a no-op and remove it from the codegen. Another way to waste time would be:

```js
let n = 0;
for (let i = 0; i < 1e6; i++) {
  n = (n * i + 1) % 1e6
}
invariant(n === 577020, "…")
```

What's important is to test `n` after the loop so we are sure the value has be computed.

Reply:
ok thanks